src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -68,33 +89,23 @@ void BindingManager::HandleDeviceConnected(void * context, OperationalDeviceProx
 
 void BindingManager::HandleDeviceConnected(OperationalDeviceProxy * device)
 {
-    mPendingNotificationMap.ForEachActiveObject([&](PendingNotificationEntry * entry) -> Loop {
-        if (entry->GetPeerId() == device->GetPeerId())
-        {
-            SyncPendingNotificationsToPeer(device, entry);
-        }
-
-        return Loop::Continue;
-    });
-}
+    FabricIndex fabricToRemove = kUndefinedFabricIndex;
+    NodeId nodeToRemove        = kUndefinedNodeId;
 
-void BindingManager::SyncPendingNotificationsToPeer(OperationalDeviceProxy * device, PendingNotificationEntry * pendingClusters)
-{
-    for (const ClusterPath & path : *pendingClusters)
+    for (const PendingNotificationEntry & pendingNotification : mPendingNotificationMap)
     {
-        ClusterId cluster   = path.cluster;
-        EndpointId endpoint = path.endpoint;
-        for (uint8_t j = 0; j < EMBER_BINDING_TABLE_SIZE; j++)
+        EmberBindingTableEntry entry;
+        emberGetBinding(pendingNotification.mBindingEntryId, &entry);
+
+        PeerId peer = PeerIdForNode(mAppServer->GetFabricTable(), entry.fabricIndex, entry.nodeId);
+        if (device->GetPeerId() == peer)
         {
-            EmberBindingTableEntry entry;
-            if (emberGetBinding(j, &entry) == EMBER_SUCCESS && entry.type == EMBER_UNICAST_BINDING && entry.clusterId == cluster &&
-                entry.local == endpoint && mBoundDeviceChangedHandler)
-            {
-                mBoundDeviceChangedHandler(&entry, device, path.context);
-            }
+            fabricToRemove = entry.fabricIndex;
+            nodeToRemove   = entry.nodeId;
+            mBoundDeviceChangedHandler(&entry, device, pendingNotification.mContext);
         }
     }
-    mPendingNotificationMap.RemoveEntry(pendingClusters);
+    mPendingNotificationMap.RemoveAllEntriesForNode(fabricToRemove, nodeToRemove);
```

Comment:
Might be simpler to just remove all entries that match the fabric/node of `device->GetPeerId()` so we don't have to save off `fabricToRemove`/`nodeToRemove`.

Reply:
The `PendingNotificationMap` stores only the fabric index. Mapping the `compressedFabricId` back to the fabric index requires iterating through the fabric table and is not supported now. Storing the index may be the simpler way.